ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestInitiator.java

```diff
@@ -1200,6 +1202,36 @@ public void resolveUserToRunAs() throws Exception {
     Assert.assertEquals(tableName, compacts.get(0).getTablename());
   }
 
+  @Test
```

Comment:
Could you add a test for the check in Worker as well?

Reply:
Done